src/core/qgslayerdefinition.cpp

```diff
@@ -274,13 +277,16 @@ QDomDocument QgsLayerDefinition::exportLayerDefinitionLayers( const QList<QgsMap
 QList<QgsMapLayer *> QgsLayerDefinition::loadLayerDefinitionLayers( QDomDocument &document, QgsReadWriteContext &context )
 {
   QList<QgsMapLayer *> layers;
-  QDomNodeList layernodes = document.elementsByTagName( QStringLiteral( "maplayer" ) );
-  for ( int i = 0; i < layernodes.size(); ++i )
+  QDomElement layerElem = document.documentElement().firstChildElement( QStringLiteral( "projectlayers" ) ).firstChildElement( QStringLiteral( "maplayer" ) );
```

Comment:
Just a note that this behaviour is slightly different- elementsByTagName searches recursively, whereas the child element approach will only find direct children. If anything though you're fixing a potential bug by ensuring that only direct children are found here though!

Reply:
Yeah, I thought about that. **In Travis we trust** for other regressions.